src/main/java/jewellery/inventory/service/SaleService.java

```diff
@@ -0,0 +1,92 @@
+package jewellery.inventory.service;
+
+import java.util.List;
+import java.util.UUID;
+import jewellery.inventory.dto.request.SaleRequestDto;
+import jewellery.inventory.dto.response.SaleResponseDto;
+import jewellery.inventory.exception.not_found.ProductNotFoundException;
+import jewellery.inventory.exception.not_found.UserNotFoundException;
+import jewellery.inventory.exception.product.ProductOwnerEqualsRecipientException;
+import jewellery.inventory.exception.product.ProductOwnerNotSeller;
+import jewellery.inventory.mapper.SaleMapper;
+import jewellery.inventory.model.Product;
+import jewellery.inventory.model.Sale;
+import jewellery.inventory.model.User;
+import jewellery.inventory.repository.ProductRepository;
+import jewellery.inventory.repository.SaleRepository;
+import jewellery.inventory.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class SaleService {
+  private final SaleRepository saleRepository;
+  private final SaleMapper saleMapper;
+  private final ProductRepository productRepository;
+  private final UserRepository userRepository;
```

Comment:
I'd prefer if we used the UserService and ProductService for User and Products fetching/updating logic because the repos provide a very bare logic for dealing with the entities. We should be handling the entities through the services. You can consult with @ignatIgnatov about the complications involved and how to address them.